do_mpc/model.py

```diff
@@ -1235,3 +1236,148 @@ def setup(self):
+        
+            .. math::
+                \\dot{x} = f(x,u,z) \\\\
+                    0 = g(x,u,z)
+       
+        where :math:`x` is the states, :math:`u` is the input and :math:`z` is the algebraic states of the system.
+        Differentiation method is as follows:
+        
+            .. math::
+                \\dot{z} = -\\frac{\\partial g}{\\partial z}^{-1}\\frac{\\partial g}{\\partial x}f-\\frac{\\partial g}{\\partial z}^{-1}\\frac{\\partial g}{\\partial u}\\dot{u}
+        
+        Therefore the converted ODE system looks like:
+        
+            .. math::
+                \\begin{pmatrix} \\dot{x} \\\\ \\dot{u} \\\\ \\dot{z} \\end{pmatrix} = \\begin{pmatrix} f(x,u,z) \\\\ q \\\\ g(x,u,z) \\end{pmatrix}
+        
+        where :math:`\\dot{x},\\dot{u},\\dot{z}` are the states of the model and q is the input to the model. Similarly, it can be extended to discrete time systems.
+        
+        :return: Converted ODE Model
+        :rtype: model.Model 
+        
+
+
+        """
+        #Check whether model setup is done
+        assert self.flags['setup'] == True, 'Run this function after original model is setup'
+
+        #Initializing new model
+        daeModel = Model(self.model_type)
+        
+        #Setting states and inputs
+        x = []
+        for i in range(self.n_x):
+            x.append(daeModel.set_variable('_x',self.x.keys()[i]))
+        for j in range(self.n_u):
+            x.append(daeModel.set_variable('_x',self.u.keys()[j+1]))
+        for k in range(self.n_z):
+            x.append(daeModel.set_variable('_x',self.z.keys()[k+1]))
+        q = daeModel.set_variable('_u','q',(self.n_u,1))
+        
+        #Extracting variables
+        x_new = daeModel.x[self.x.keys()]
+        u_new = daeModel.x[self.u.keys()[1:]]
+        z_new = daeModel.x[self.z.keys()[1:]]
+        tvp_new = daeModel.tvp[self.tvp.keys()]
+        p_new = daeModel.p[self.p.keys()]
+        w_new = daeModel.w[self.w.keys()]
+        
+        #Converting rhs eq. with respect to variables of linear model of same name
+        rhs = self._rhs_fun(vertcat(*x_new),vertcat(*u_new),vertcat(*z_new),vertcat(*tvp_new),vertcat(*p_new),vertcat(*w_new))
+        alg = self._alg_fun(vertcat(*x_new),vertcat(*u_new),vertcat(*z_new),vertcat(*tvp_new),vertcat(*p_new),vertcat(*w_new))
+        z_next = -inv(jacobian(alg,vertcat(*z_new)))@jacobian(alg,vertcat(*x_new))@rhs-inv(jacobian(alg,vertcat(*z_new)))@jacobian(alg,vertcat(*u_new))@q
+        u_next = q
+        
+        #Computing rhs of the model
+        for i in range(self.n_x):
+            daeModel.set_rhs(self.x.keys()[i],rhs[i])
+        for j in range(self.n_u):
+            daeModel.set_rhs(self.u.keys()[j+1],daeModel.u['q',j])
+        for k in range(self.n_z):
+            daeModel.set_rhs(self.z.keys()[k+1],z_next[k])
+        
+        #setting up the model
+        daeModel.setup()
+        daeModel.flags['dae2odemodel'] = True
```

Comment:
What is this used for?

Reply:
This flag indicates that the model was converted from dae to ode. So then accordingly cost matrix can be modified.